test/jdk/jdk/crac/LazyProps.java

```diff
@@ -22,34 +22,39 @@
  */
 
 import jdk.crac.*;
-import jdk.test.lib.process.ProcessTools;
-import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.crac.CracBuilder;
+import jdk.test.lib.crac.CracEngine;
+import jdk.test.lib.crac.CracTest;
 
 /**
  * @test
  * @library /test/lib
  * @run main LazyProps
  */
-public class LazyProps {
-    static public void main(String[] args) throws Exception {
-        if (args.length == 0) {
-            OutputAnalyzer output = ProcessTools.executeTestJvm(
-                    "-XX:CREngine=simengine", "-XX:CRaCCheckpointTo=./cr",
-                    "LazyProps",
-                    "-runTest");
-            output.shouldHaveExitValue(0);
-            output.shouldContain("jdk.crac beforeCheckpoint");
-        } else {
-            Resource resource = new Resource() {
-                @Override
-                public void beforeCheckpoint(Context<? extends Resource> context) throws Exception { }
-                @Override
-                public void afterRestore(Context<? extends Resource> context) throws Exception { }
-            };
-            Core.getGlobalContext().register(resource);
+public class LazyProps implements CracTest {
+    public static void main(String[] args) throws Exception {
+        CracTest.run(LazyProps.class, args);
+    }
+
+    @Override
+    public void test() throws Exception {
+        new CracBuilder().engine(CracEngine.SIMULATE).main(LazyProps.class).args(CracTest.args())
```

Comment:
`main(LazyProps.class).args(CracTest.args())` seems to be repeated, would it harm to make them default?

Reply:
I kind of thought about not having CracBuilder and CracTest interdependent but it's probably ok (and reduces boilerplate), so I'll do that.